pkg/entity/policy/cmd.go

```diff
@@ -0,0 +1,189 @@
+		Args:              cobra.MaximumNArgs(1), // the contract reference
+		ValidArgsFunction: namesCompletion,
+	}
+	flags := command.Flags()
+	flags.StringP(common.OutputFormatFlag, common.OutputFormatFlagShort, common.OutputFormatPlain,
+		fmt.Sprintf("output format [%v]", outputFormatFlagAllowedValuesText))
+	flags.String(idFlag, "", "policy id")
+	_ = command.RegisterFlagCompletionFunc(idFlag, idsCompletion)
+	return command
+}
+
+func CreateCmd() *cobra.Command {
+	outputFormatFlagAllowedValues := []string{common.OutputFormatPlain,
+		common.OutputFormatJson, common.OutputFormatJsonRaw}
+	outputFormatFlagAllowedValuesText := strings.Join(outputFormatFlagAllowedValues, ", ")
+	command := &cobra.Command{
+		Use:   "policy",
+		Short: "Create a Policy",
+		Long: `Create a Policy
+
+A policy has the following attributes
+* name: the name of a policy. This must be unique within one organization.
+* description: a description of the policy; what sort of data pipelines
+  would be subject to this policy?
+* retention: the number of days that encryption keys created under this
+  policy should be kept. This might be a minimum or a maximum...
+* legal grounds: a legal text or ruling that identifies why the organization
+  created this policy
+* state: draft, active or archived. Policies can only be used in pipelines
+  when they're in active state. They can still be modified while in draft.
+  Deletion of policies is not allowed for active policies.
+`,
+		Example:           `strm create policy --name="1 year" --retention 365 --description "1 year for marketing"`,
+		DisableAutoGenTag: true,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			printer = configurePrinter(cmd)
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			create(cmd)
+		},
+		Args: cobra.NoArgs,
+	}
+	flags := command.Flags()
+	flags.StringP(common.OutputFormatFlag, common.OutputFormatFlagShort, common.OutputFormatPlain,
+		fmt.Sprintf("output format [%v]", outputFormatFlagAllowedValuesText))
+	flags.String(nameFlag, "", "name")
+	flags.String(descriptionFlag, "", "description of the policy")
+	flags.String(legalGroundsFlag, "", "legal grounds of this policy")
+	flags.Int32(retentionFlag, 365, "retention in days of this policy")
+	flags.String(stateFlag, "draft", "State of the policy. Choose from draft, active, archived")
+	_ = command.RegisterFlagCompletionFunc(stateFlag, stateCompletion)
+	_ = command.MarkFlagRequired(nameFlag)
+	_ = command.MarkFlagRequired(retentionFlag)
+	return command
+}
+
+func UpdateCmd() *cobra.Command {
+	outputFormatFlagAllowedValues := []string{common.OutputFormatPlain,
+		common.OutputFormatJson, common.OutputFormatJsonRaw}
+	outputFormatFlagAllowedValuesText := strings.Join(outputFormatFlagAllowedValues, ", ")
+	command := &cobra.Command{
+		Use:     "policy policy-id",
+		Short:   "Update a Policy",
+		Example: `strm update policy 34c4709e-b8bc-4b45-aa5a-883f471869e3 --state active --legal-grounds "all your base are belong to us"`,
```

Comment:
The example mentions `--state active`, which implies that we're activating the policy to the active state. We have a dedicated verb `activate` in the CLI to change entities from state. It might be better to disallow updating the state in this update call and changing state through the dedicated verbs.

Reply:
implemented (same for archive)

Reply:
So the `--state active` can be removed from the example, right? @bvdeenen